### PR TITLE
Add 16bit register support

### PR DESCRIPTION
**Pull request by MadisonC-SparkFun**
This will add support for 16-bit register addresses for SPI and I2C implementations. This code has been tested on the XM125 16-bit address, 32-bit registers. The specific functions added for I2C and SPI: 

- readRegister16Region(uint16_t reg, uint8_t *data, size_t numBytes) 
- writeRegister16Region(uint16_t devReg, uint8_t *data, size_t length)

### CODE REVIEW

File: src/sfeTkArdI2C.cpp [hunk 1]
```diff
@@ -238,7 +213,10 @@ sfeTkError_t sfeTkArdI2C::readRegisterRegion(uint8_t devReg, uint8_t *data, size
         {
             _i2cPort->beginTransmission(address());
 
-            _i2cPort->write(devReg);
+            // uint16_t *foo = (uint16_t*)devReg;
```
Comment (gigapod) [inline src/sfeTkArdI2C.cpp:216]: Remove old / unused code

File: src/sfeTkArdI2C.h [hunk 2]
```diff
@@ -211,6 +237,10 @@ class sfeTkArdI2C : public sfeTkII2C
     TwoWire *_i2cPort;
 
   private:
+sfeTkError_t writeRegisterRegionAddress(uint8_t *devReg, size_t regLength, const uint8_t *data, size_t length);
```
Comment (gigapod) [inline src/sfeTkArdI2C.h:240]: formatting -> these shoudl be indented and align with above methods. Run the file through the formatter in your editor.  Details here: https://github.com/sparkfun/.github/blob/main/SFECodeStyleGuide.md

**Review by gigapod — APPROVED** [review]
Looks good. Just a couple of nits to clean up -- see comments

File: src/sfeTkArdI2C.h (hunk 2 above)
Comment (MadisonC-SparkFun) [reply src/sfeTkArdI2C.h]: Functions have been indented and are now aligned

File: src/sfeTkArdI2C.cpp (hunk 1 above)
Comment (MadisonC-SparkFun) [reply src/sfeTkArdI2C.cpp]: Unneeded code has been deleted

**Pull request merged**